### PR TITLE
Use the LanguageTool server for improved performance

### LINKED ISSUE #6: Poor performance of LanguageTool command line checker
I find that my CPU is unusually busy when using flycheck-languagetool.

I notice that `languagetool-commandline.jar` takes a long time to start up, even if there is no work for it to do.  We pay this cost every time the check runs.

```console
$ time java -jar languagetool-commandline.jar /dev/null
No language specified, using English (no spell checking active, specify a language variant like 'en-GB' if available)
Working on /dev/null...
Time: 76ms for 0 sentences (0.0 sentences/sec)

real    0m2.838s
user    0m11.031s
sys     0m0.255s
```

If we were to use `languagetool-server.jar` instead, subsequent checks would be much faster:

```console
$ java -cp languagetool-server.jar org.languagetool.server.HTTPServer
```
```console
$ time curl --data "language=en-GB&text=" http://localhost:8081/v2/check
{"software":{"name":"LanguageTool","version":"5.3","buildDate":"2021-03-29 11:16:13 +0000","apiVersion":1,"premium":false,"premiumHint":"You might be missing errors only the Premium version can find. Contact us at support<at>languagetoolplus.com.","status":""},"warnings":{"incompleteResults":false},"language":{"name":"English (GB)","code":"en-GB","detectedLanguage":{"name":"English (US)","code":"en-US","confidence":0.0}},"matches":[]}
real    0m2.365s
user    0m0.005s
sys     0m0.007s
$ time curl --data "language=en-GB&text=" http://localhost:8081/v2/check
{"software":{"name":"LanguageTool","version":"5.3","buildDate":"2021-03-29 11:16:13 +0000","apiVersion":1,"premium":false,"premiumHint":"You might be missing errors only the Premium version can find. Contact us at support<at>languagetoolplus.com.","status":""},"warnings":{"incompleteResults":false},"language":{"name":"English (GB)","code":"en-GB","detectedLanguage":{"name":"English (US)","code":"en-US","confidence":0.0}},"matches":[]}
real    0m0.064s
user    0m0.006s
sys     0m0.006s

```

### PR DESCRIPTION
**Pull request by mavit**
I got a bit carried away, here.  Apologies for such a large pull request.

Fixes #6 and obsoletes #5.

### CODE REVIEW

File: flycheck-languagetool.el [hunk 1]
```diff
@@ -1,12 +1,13 @@
 ;;; flycheck-languagetool.el --- Flycheck support for LanguageTool  -*- lexical-binding: t; -*-
 
-;; Copyright (C) 2021  Shen, Jen-Chieh
+;; Copyright (C) 2021  Shen, Jen-Chieh; Peter Oliver
 ;; Created date 2021-04-02 23:22:44
 
 ;; Author: Shen, Jen-Chieh <jcs090218@gmail.com>
+;;	Peter Oliver <git@mavit.org.uk>
```
Comment (jcs090218) [inline flycheck-languagetool.el]: Can you replace tab with spaces? Thanks!

File: flycheck-languagetool.el [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 ;; Description: Flycheck support for LanguageTool.
 ;; Keyword: grammar check
-;; Version: 0.2.0
+;; Version: 0.3.0
 ;; Package-Requires: ((emacs "25.1") (flycheck "0.14") (s "1.9.0"))
```
Comment (jcs090218) [inline flycheck-languagetool.el]: I think we no longer need `(s "1.9.0")` as dependencies?

**Review by jcs090218 — APPROVED** [review]
Small nit, can you fix these for me? Thanks!

Otherwise looks good! :)

**Comment by mavit** [comment]
Both issues fixed.  Thanks.

**Comment by jcs090218** [comment]
Looks good! Thanks! 😄 👍

**Pull request merged**